app/marketing/views.py

```diff
@@ -885,6 +886,25 @@ def trending_quests():
         ).order_by('-recent_attempts').all()[0:10]
     return quests
 
+def quest_of_the_day():
+    quest = trending_quests()[0]
+    return quest
+
+def upcoming_grant():
+    cutoff_date = timezone.now() - timezone.timedelta(days=7)
+    grant = Grant.objects.filter(created_on__gte=cutoff_date).order_by('-created_on')[:5][0]
+    return grant
+
+def upcoming_hackathon():
+    cutoff_date = timezone.now() - timezone.timedelta(days=7)
+    hackathon = HackathonEvent.objects.filter(created_on__gte=cutoff_date).order_by('-created_on')[:5][0]
```

Comment:
order by start date, and choose one thats going on now or is coming up soon

Reply:
Ok. Will try doing that.

Reply:
Will this do?:

```
def upcoming_hackathon():
    try:
        return HackathonEvent.objects.filter(end_date__gt=timezone.now()).order_by('-start_date')[:5][0]
    except HackathonEvent.DoesNotExist:
        try:
            return HackathonEvent.objects.filter(start_date__gte=timezone.now()).order_by('start_date')[:5][0]
        except HackathonEvent.DoesNotExist:
            return None
```

Reply:
I shall use .first(). 

I think the code above does get the hackathons that are upcoming or going on now.

I believe, 

`.filter(end_date__gt=timezone.now()).order_by('-start_date').first()` gets the latest ongoing hackathon and if there isn't anything ongoing,
`.filter(start_date__gte=timezone.now()).order_by('start_date').first()` gets the next upcoming hackathon.

Reply:
i think youll want to get all hackathons going on now. yes that looks like right for doing just one hackathon

Reply:
You mean, all the ongoing and/or upcoming hackathons should be displayed in the email? not just `.first()`?